app/src/main/java/com/example/habitsmasher/DatabaseEntity.java

```diff
@@ -16,23 +21,32 @@
      * Constructor for database entity
      * @param id id of database entity
      */
-    DatabaseEntity(long id) {
+    DatabaseEntity(String id) {
         _id = id;
     }
 
     /**
      * Sets the ID of the database entity
      * @param id new ID of the entity
      */
-    private void setId(long id) {
+    private void setId(String id) {
         _id = id;
     }
 
     /**
      * Gets the ID of the database entity
      * @return the ID of the entity
      */
-    public long getId() {
+    public String getId() {
         return _id;
     }
+
+    /**
+     * Generates a random UUID in the form of a string for the
+     * database entity
+     * @return
+     */
+    public static String generateId() {
+        return UUID.randomUUID().toString();
+    }
 }
```

Comment:
Method used to generate new IDs for any database entity wrapped into this static method